frontend-react/docs/best-practices.md

```diff
@@ -0,0 +1,677 @@
+
+Keep in mind that with any other guidelines, there are always going to be exceptions.  In these cases, use your best judgment, talk to your teammates, and make sure to note historical reasoning via inline or review comments.
+
+## Guiding Principles
+
+### Effort is not measured by lines of code
+
+> Why?
+>
+> Lines of code committed is an unreliable metric for the amount of effort put into any particular changeset.  Essentially, this comes down to a "quality versus quantity" argument, but much more eloquent authors have penned articles like [this](https://www.pluralsight.com/blog/teams/lines-of-code-is-a-worthless-metric--except-when-it-isn-t-) and [this](https://www.gitclear.com/count_lines_of_code_at_your_own_peril) about the topic.
+
+### Strive for reusability; don't reinvent the wheel
+
+> Why?
+>
+> In many cases, there are already vetted, well-tested, and well-maintained libraries out there that can assist us in accomplishing our goals.  When considering building our own custom solutions, we also need to factor in the long-term costs of maintenance (e.g., testing, versioning, augmenting) before we make a decision.  It may very well be the case that we _do_ need a custom solution, but it's always worth doing some investigation to see if something that already exists fits the bill.
+
+### Don't let perfection be the enemy of done
+
+> Why?
+>
+> It goes without saying, but no codebase or approach will ever be perfect; users and requirements will change, and it's arguably more important to leave space for flexibility and extensibility so that we can respond accordingly to where the future takes us.
+>
+> That said, it's always important to weigh options and other teams' bandwidths properly.  Are "suboptimal" solutions good enough for now or are there hard blockers or issues?
+
+### Opt for clarity over cleverness; avoid premature optimizations
+
+> Why?
+> 
+> The goal of our software is long-term scalability, which involves multiple humans reading and working in the codebase at any given time.  Regardless of how familiar we may be with the underlying technologies, patterns, and processes we use, code becomes legacy as soon as it's merged into the master branch; it no longer "belongs" to any individual person because another person may need to work on it the next day.  Therefore, it's one of our jobs as a team to reduce the barrier to (re-)entry for each other and ourselves down the line.  
+> 
+> Read more about this [here](https://thedailywtf.com/articles/Programming-Sucks!-Or-At-Least%2c-It-Ought-To-) and [here](https://www.joshwcomeau.com/career/clever-code-considered-harmful/).
+
+---
+
+## Processes
+
+### Branch names
+
+Branch names should follow the format:
+
+```  
+experience/XXXX/some-short-description-of-the-feature  
```

Comment:
Should just leave a little comment here what `XXXX` means (the Zenhub ticket number).

Reply:
Sure, can do!